### DISCUSSION TITLE
Reboot during install, getting stuck at apfs_module_stop:1730

### DISCUSSION THREAD

**Discussion by Metrabyte** [CodeRunner5235/Opencore-Steam-Deck#5] [Q&A] [answered]
Hi there! First of all, thank you so much for this guide! I'm installing Catalina 10.15.7 (trying to use an offline installer by following: https://www.reddit.com/r/hackintosh/comments/jrrhox/how_to_make_a_full_offline_installer_for_macos_on/) on my Steam Deck LCD, with the 512GB nvme drive. I've gotten it to boot and start the installer successfully. After it does the first reboot into the installer, it hangs at an "apfs_module_stop" line and "(Remaining_mem_allocated: 0)". Attached are my EFI file and a picture of the screen, along with a log. Thank you for any help :)
[EFI.zip](https://github.com/user-attachments/files/17457017/EFI.zip)
[opencore-2024-10-21-032915.txt](https://github.com/user-attachments/files/17457018/opencore-2024-10-21-032915.txt)

![IMG_E6824](https://github.com/user-attachments/assets/a61cad52-1608-4b11-8427-d4a9016272a3)

**Comment by Metrabyte** [comment]
Oh also, to add, I tried the EFI file from this thread: https://github.com/CodeRunner5235/Opencore-Steam-Deck/discussions/1#discussioncomment-10885512 from the discussions and it didn’t work as well. It actually made it to where my usb devices wouldn’t map properly as it used the ryzen fix kext for Big Sur and above. I replaced that kext and got back the same error. 

I’ve also tried formatting as both APFs and MacOS extended Journaled. It didn’t seem to have much of an effect.

**Comment by CodeRunner5235** [comment]
Looking through your config.plist, I noticed that your MinVersion and MinDate under UEFI -> APFS are set for Catalina. Try setting them both to no restriction ( a -1 in both) and see if that fixes the issue 
![image](https://github.com/user-attachments/assets/9fdbae76-4fb9-4419-b883-d0f88168b432)
![image](https://github.com/user-attachments/assets/70ace33a-8e16-4ab8-bbfd-f56c15014510)

**Reply by Metrabyte** [reply]
I actually had it like that before and thought setting it to the proper Catalina values you see there would fix it, but unfortunately not. Thank you for your response though! Any other possible ideas?

**Reply by CodeRunner5235** [reply]
Some more poking around and comparing, I have some more stuff to try. set  Booter -> Quirks -> FixupAppleEfiImages to false  Booter -> Quirks -> SetupVirtualMap to true and UEFI -> ProtocolOverrides -> AppleEg2Info to true.

And this is me comparing your plist to a known good one that I have laying around

**Reply by Metrabyte** [reply]
Unfortunately that didn't work either but thank you again! I didn't even have Booter -> Quirks -> FixupAppleEfiImages at all so I added it and set it to false. Still nothing sadly. I'm thinking something could be wrong with my offline installer or something.. I may try a different version of MacOS.

**Reply by CodeRunner5235** [reply]
What version of opencore are you using? What bios version on the deck? Catalina is easier to boot and newer versions, you can try removing the vsmcgen from your boot args to see if that helps catalina (dont know why it would). Or you can try with Big Sur or newer, that is what is known to work well with this configuration.

**Reply by Metrabyte** [reply]
I'm on Opencore 1.0.2 and BIOS version F7A0120. I was trying Catalina since I thought it was the simpler version lol. I'll try removing vsmcgen and see what that does. If not, I'll try Big Sur and hope for the best!

**Reply by Metrabyte** [reply, marked as answer]
I setup an offline installer of Big Sur instead, using MacOS in a VM and got everything setup, including the known working EFI. It got most of the way through the installer and then hung on "ifnet attach: All kernel threads created for interface lo0 have been scheduled at least once. Procee" and it cuts off right there. I let it sit for about 15 minutes and still nothing. So certainly further, but not actually getting into macOS unfortunately.

Edit: I restarted it and it booted.

**Reply by Metrabyte** [reply]
It booted! I literally just restarted it and it completely installed now. No changes to EFI or BIOS as well. Anyway, thank you for all your guidance!

**Reply by CodeRunner5235** [reply]
No problem! There is some kind of racing condition (we think) so if it hangs on boot, give it a restart and that should fix it
